sdk/storage/azblob/appendblob/client.go

```diff
@@ -36,11 +37,14 @@ func NewClient(blobURL string, cred azcore.TokenCredential, options *ClientOptio
 	authPolicy := shared.NewStorageChallengePolicy(cred)
 	conOptions := shared.GetClientOptions(options)
 	conOptions.PerRetryPolicies = append(conOptions.PerRetryPolicies, authPolicy)
```

Comment:
Now that `authPolicy` is in the pipeline options you should remove it from here (comment applies to all constructors where applicable).